### PR TITLE
Fix grouped range queries dropping field from parent GroupNode

### PR DESCRIPTION
**Pull request by niemyjski**
## Summary

- Fix \TermRangeNode\s inside a \GroupNode\ (e.g., \ield:(>30 AND <=40)\) having their range clauses silently dropped because they had no explicit field set
- Propagate the parent \GroupNode\'s field to child \TermRangeNode\s when the child has no field
- Add 3 new tests validating grouped range query generation
- Bump \Elastic.Clients.Elasticsearch\ from 8.19.21 to 8.19.22

## Root Cause

The Lucene query parser has always created \TermRangeNode\s without their own \Field\ when inside a \GroupNode\ (e.g., \ield:(>30 AND <=40)\ produces a \GroupNode\ with field=\ield\ containing two \TermRangeNode\s with field=null).

In the v7/NEST era, \GetDefaultQueryAsync\ would still create a \TermRangeQuery { Field = null }\, and NEST would silently drop it during serialization — so the query was always broken, but it didn't throw.

After the v8 upgrade (d91818b), the \Elastic.Clients.Elasticsearch\ \TermRangeQuery\ constructor requires a non-null field, so \GetDefaultQueryAsync\ was changed to \
eturn null\ when the field is empty. This made the bug explicit: range clauses inside grouped fields were now completely dropped.

## Fix

In \DefaultQueryNodeExtensions.GetDefaultQueryAsync(TermRangeNode)\, when \
ode.UnescapedField\ is empty, we now check if the parent is a \GroupNode\ with a field and propagate it before falling through to the existing \
eturn null\ path.

## Test plan

- [x] \GroupedRangeQueryPropagatesToChildTermRangeNodes\ - verifies \ield:(>30 AND <=40)\ produces two \TermRangeQuery\s with correct field and bounds  
- [x] \GroupedRangeQueryWithInclusiveBoundaries\ - verifies \ield:(>=10 AND <20)\ with gte/lt operators
- [x] \GroupedRangeQueryCombinedWithOtherTerms\ - verifies combined query \ield:(>30 AND <=40) AND other:value\ produces a valid bool query

### CODE REVIEW

**Comment by niemyjski** [comment]
All review feedback has been addressed in commit 9901e29:

1. **Null-safe cast pattern**: Replaced all `as TermRangeQuery` + `Assert.NotNull` with `Assert.IsType<TermRangeQuery>(...)` which returns a non-null typed value and provides better failure messages.

2. **Tests moved to correct class**: Moved all 3 test methods out of the nested `UpdateFixedTermFieldToDateFixedExistsQueryVisitor` helper class into the main `ElasticQueryParserTests` class.

3. **Strengthened shallow test**: `BuildQueryAsync_GroupedRangeWithOtherTerms_ProducesBoolQueryWithAllClauses` now asserts clause count, verifies 2 TermRangeQuery clauses targeting "field", and 1 term query targeting "other".

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 1]
```diff
@@ -1713,6 +1714,78 @@ public override void Visit(TermNode node, IQueryVisitorContext context)
...
+        Assert.NotNull(result);
+        var boolQuery = result.Bool;
+        Assert.NotNull(boolQuery);
+        var clauses = boolQuery.Filter ?? boolQuery.Must;
+        Assert.NotNull(clauses);
+        Assert.Equal(2, clauses.Count);
+
+        var termRange1 = clauses.ElementAt(0).Range as TermRangeQuery;
+        Assert.NotNull(termRange1);
+        Assert.Equal("field", termRange1.Field.ToString());
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs]: Fixed in 9901e29 — replaced `var termRangeX = ... as TermRangeQuery; Assert.NotNull(termRangeX);` with `var termRangeX = Assert.IsType<TermRangeQuery>(...);` which both eliminates the null warning and strengthens the assertion.

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 2]
```diff
@@ -1713,6 +1714,78 @@ public override void Visit(TermNode node, IQueryVisitorContext context)
...
+        Assert.Equal("30", termRange1.Gt);
+
+        var termRange2 = clauses.ElementAt(1).Range as TermRangeQuery;
+        Assert.NotNull(termRange2);
+        Assert.Equal("field", termRange2.Field.ToString());
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs]: Fixed in 9901e29 — same `Assert.IsType<TermRangeQuery>()` refactoring applied here.

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 3]
```diff
@@ -1713,6 +1714,78 @@ public override void Visit(TermNode node, IQueryVisitorContext context)
...
+        Assert.NotNull(result);
+        var boolQuery = result.Bool;
+        Assert.NotNull(boolQuery);
+        var clauses = boolQuery.Filter ?? boolQuery.Must;
+        Assert.NotNull(clauses);
+        Assert.Equal(2, clauses.Count);
+
+        var termRange1 = clauses.ElementAt(0).Range as TermRangeQuery;
+        Assert.NotNull(termRange1);
+        Assert.Equal("field", termRange1.Field.ToString());
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs]: Fixed in 9901e29 — same `Assert.IsType<TermRangeQuery>()` refactoring applied here.

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 4]
```diff
@@ -1713,6 +1714,78 @@ public override void Visit(TermNode node, IQueryVisitorContext context)
...
+        Assert.Equal("10", termRange1.Gte);
+
+        var termRange2 = clauses.ElementAt(1).Range as TermRangeQuery;
+        Assert.NotNull(termRange2);
+        Assert.Equal("field", termRange2.Field.ToString());
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs]: Fixed in 9901e29 — same `Assert.IsType<TermRangeQuery>()` refactoring applied here.

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 5]
```diff
@@ -1713,6 +1714,78 @@ public override void Visit(TermNode node, IQueryVisitorContext context)
         var query = new Query { Exists = new ExistsQuery { Field = "date_fixed" } };
         node.SetQuery(isFixed ? query : !query);
     }
+
+    [Fact]
+    public async Task BuildQueryAsync_GroupedRangeWithExclusiveBounds_PropagatesFieldToTermRangeNodes()
+    {
+        // Arrange
+        var sut = new ElasticQueryParser();
+
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs:1685]: Fixed in 9901e29 — the test methods were moved out of the helper visitor class and into the main `ElasticQueryParserTests` class body.

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 6]
```diff
@@ -1713,6 +1714,78 @@ public override void Visit(TermNode node, IQueryVisitorContext context)
...
+        // Act
+        var result = await sut.BuildQueryAsync("field:(>30 AND <=40) AND other:value");
+
+        // Assert
+        Assert.NotNull(result);
+        var boolQuery = result.Bool;
+        Assert.NotNull(boolQuery);
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs:1746]: Fixed in 9901e29 — strengthened assertions to verify clause count and types (2 TermRangeQuery + 1 TermQuery). Further strengthened in b5a5972 to locate clauses by bound values rather than position.

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 7]
```diff
@@ -1675,6 +1676,88 @@ public async Task CanValidateAggregation(string aggregation)
...
+        Assert.NotNull(clauses);
+        Assert.True(clauses.Count >= 3, $"Expected at least 3 clauses but found {clauses.Count}");
+
+        var rangeQueries = clauses.Where(c => c.Range is TermRangeQuery).ToList();
+        Assert.Equal(2, rangeQueries.Count);
+        foreach (var rq in rangeQueries)
+        {
+            var termRange = Assert.IsType<TermRangeQuery>(rq.Range);
+            Assert.Equal("field", termRange.Field.ToString());
+        }
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs]: Addressed in b5a5972 — replaced the `foreach` assertion loop with `Assert.All()` which is the idiomatic xUnit pattern for asserting a condition across all items in a collection. Using `.Select()` would be semantically wrong since the loop performs assertions (side effects), not transformations.

File: src/Foundatio.Parsers.ElasticQueries/Extensions/DefaultQueryNodeExtensions.cs [hunk 8]
```diff
@@ -345,7 +345,12 @@ private static async Task<Query> GetSplitNestedQueryAsync(TermNode node, Diction
 
         string? field = node.UnescapedField;
         if (String.IsNullOrEmpty(field))
-            return null;
+        {
+            if (node.Parent is GroupNode parentGroup && !String.IsNullOrEmpty(parentGroup.UnescapedField))
+                field = parentGroup.UnescapedField;
+            else
+                return null;
+        }
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply src/Foundatio.Parsers.ElasticQueries/Extensions/DefaultQueryNodeExtensions.cs:353]: Acknowledged — nested parentheses like `field:((>30) AND <=40)` would create intermediate GroupNode layers. This PR intentionally fixes the common single-level case. The nested-paren scenario is an edge case we can track separately if it surfaces as a real issue. The parser already handles most practical grouped range expressions.

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 9]
```diff
@@ -1675,6 +1676,88 @@ public async Task CanValidateAggregation(string aggregation)
...
+        Assert.NotNull(clauses);
+        Assert.Equal(2, clauses.Count);
+
+        var termRange1 = Assert.IsType<TermRangeQuery>(clauses.ElementAt(0).Range);
+        Assert.Equal("field", termRange1.Field.ToString());
+        Assert.Equal("30", termRange1.Gt);
+
+        var termRange2 = Assert.IsType<TermRangeQuery>(clauses.ElementAt(1).Range);
+        Assert.Equal("field", termRange2.Field.ToString());
+        Assert.Equal("40", termRange2.Lte);
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs]: Fixed in b5a5972 — refactored to locate clauses by bound values (`r.Gt`, `r.Gte`, `r.Lt`, `r.Lte`) instead of positional `ElementAt()`. Tests are now order-independent.

File: tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs [hunk 10]
```diff
@@ -1675,6 +1676,88 @@ public async Task CanValidateAggregation(string aggregation)
...
+        Assert.NotNull(clauses);
+        Assert.Equal(2, clauses.Count);
+
+        var termRange1 = Assert.IsType<TermRangeQuery>(clauses.ElementAt(0).Range);
+        Assert.Equal("field", termRange1.Field.ToString());
+        Assert.Equal("10", termRange1.Gte);
+
+        var termRange2 = Assert.IsType<TermRangeQuery>(clauses.ElementAt(1).Range);
+        Assert.Equal("field", termRange2.Field.ToString());
+        Assert.Equal("20", termRange2.Lt);
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply tests/Foundatio.Parsers.ElasticQueries.Tests/ElasticQueryParserTests.cs]: Fixed in b5a5972 — same refactoring applied here. Clauses are now located by their bound values (Gte/Lt) rather than position.
